File: cpp/Dynamic Programming/minimum_sum_partition.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
int dp[1000][1000];
int f(int n,int i,int a[],int sm,int s)
{
    if(i == n)
    {
        return abs(s-2*sm)
    }
    if(dp[i][sm] != -1)
    {
        return dp[i][sm];
    }
    return min(f(n,i+1,a,sm+a[i],s),f(n,i+1,a,sm,s));
}
int main() {
    int t;cin>>t;
    while(t--)
    {
        int n;cin>>n;
        int a[n];
        int sum = 0;
        for(int i = 0 ; i < n; i++)
        {
            cin>>a[i];
            sum += a[i];
        }
        memset(dp,-1,sizeof(dp));
        cout<<f(n,0,a,0,s)<<endl;
    }
    return 0;
}

File: python/minimum_sum_partition.py
def f(n,i,sm,a,s,dp):
	if i == n:
		return abs(s - 2*sm)
	if dp[i][sm] != -1:
		return dp[t]
	dp[i][sm] = min(f(n,i+1,sm+a[i],a,s,dp),f(n,i+1,sm,a,s,dp))
	return dp[i][sm]
def main():
	for _ in range(int(input())):
		n = int(input())
		a = list(map(int,input().split()))
		s = sum(a)
		dp = [list([-1]*(s+1)) for _ in range(n+1)]
		print(f(n,0,0,a,s,dp))
		# print(min(map(lambda x:abs(s-2*x),dp.values())))
main()